src/NUnitFramework/framework/Internal/TestParameters.cs

```diff
@@ -40,6 +40,7 @@ public abstract class TestParameters : ITestData, IApplyToTest
         public TestParameters()
         {
             RunState = RunState.Runnable;
+            InitializeArguments(new object[0]);
```

Comment:
This is the change causing ExpectedResultDoesNotBlockApplyToTestAttributes to fail.

Reply:
And this is the conflict:

https://github.com/nunit/nunit/blob/22ea163f1dbbecdb71b88270434c948f6e381bbd/src/NUnitFramework/framework/Internal/Builders/NUnitTestCaseBuilder.cs#L53-L65

Reply:
parms.Arguments == null is being used as in indirect way to tell whether to overwrite `parms` properties with method-level attributes. There needs to either be a direct way or the merge order needs to be swapped.

Reply:
`parms.Arguments.Length == 0` seems to satisfy all the tests, so I'll leave it at that for simplicity. I guess that would have been the original code except for the inconsistent behavior of some TestParameters constructors leaving Arguments null.